1_fetch/src/get_nwis_data.R

```diff
@@ -109,47 +109,36 @@ get_peak_flow_log <- function(files_in, file_out) {
 prescreen_daily_data <- function(filename, prov_rm = TRUE){
   #loads data from file and removes provisional and estimated data if prov_rm = TRUE
   message('loading ', filename)
-  ##Handle sites with strange column names
-  if (length(grep(pattern = '01011500', filename)) +
-      length(grep(pattern = '02196000', filename)) + 
-      length(grep(pattern = '03213000', filename)) + 
-      length(grep(pattern = '12010000', filename)) > 0){
+  d0 <- read_csv(filename,
+                 name_repair = "minimal", 
+                 col_types = cols(agency_cd = col_character(), 
+                                  site_no = col_character(), Date = col_date(format = "%Y-%m-%d"), 
+                                  discharge = col_number(), discharge_cd = col_character(), 
+                                  dateTime = col_character(), tz_cd = col_character())) %>%
+    na.omit() %>%
+    suppressWarnings() %>% 
+    suppressMessages()
+  ##Handle sites with unexpected number of columns ( != 5 )
+  if (ncol(d0) == 7) {
     #site has 2 columns named discharge and 2 named discharge_cd
-    d1 <- read_csv(filename,
-                   col_types = cols(agency_cd = col_character(),
-                                  site_no = col_character(), Date = col_date(format = "%Y-%m-%d"),
-                                  discharge = col_double(), discharge_cd = col_character(), 
-                                  discharge_1 = col_double(), discharge_cd_1 = col_character())) %>%
-      select(1:5) %>%
-      na.omit() %>%
-      suppressWarnings() %>% 
-      suppressMessages()
-    d2 <- read_csv(filename,
-                   col_types = cols(agency_cd = col_character(),
-                                    site_no = col_character(), Date = col_date(format = "%Y-%m-%d"),
-                                    discharge = col_double(), discharge_cd = col_character(), 
-                                    discharge_1 = col_double(), discharge_cd_1 = col_character())) %>%
-      select(1:3, 6:7) %>%
-      na.omit() %>%
-      suppressWarnings() %>% 
-      suppressMessages() %>%
-      rename(discharge = discharge_1, discharge_cd = discharge_cd_1)
+    d1 <- d0 %>% select(1:5) 
+    d2 <- d0 %>% select(1:3, 6:7)
```

Comment:
The na.omit has to be moved to after the `select(1:5)`. Otherwise, data will be deleted. Example: if discharge column 1 has NA and discharge column 2 doesn't, the whole row will be removed by na.omit.

```suggestion
    d1 <- d0 %>% select(1:5) %>% na.omit()
    d2 <- d0 %>% select(1:3, 6:7) %>% na.omit()
```

Reply:
Got it, changed.